examples/08_scaled_dot_product_attention/scaled_dot_product_attention.cpp

```diff
@@ -220,6 +255,11 @@ void sdp_fwd_run(uint32_t iter) {
     constexpr uint32_t subgroup_range_m = wg_tile_m_qk / sg_tile_m_qk;
     constexpr uint32_t subgroup_range_n = wg_tile_n_qk / sg_tile_n_qk;
 
+    constexpr uint32_t slm_size
+            = wg_tile_m_qk * wg_tile_n_qk * sizeof(dtype_sfx);
+    static_assert(slm_size <= arch_attr_t<arch_tag>::local_mem_size,
```

Comment:
I remember this is checked somewhere else.

Reply:
Yes. The size can be checked inside `slm_barrier_init<gemm_op_t>()` (e.g. example 01, where `gemm_op_t` should be a GEMM functor consists of prologue & gemm & epilogue). However, the sdp example uses `gemm_t` (with `default_gemm_selector_t`) directly and calculate the slm size manually (probably because sdp have a different parallel paradigm from gemm).

Therefore, explicit slm size checking is necessary here.